File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-1_Gestion-Salaires-Employes/Implementation-Cpp/Employe.cpp
#include "Employe.h"

Employe::Employe(string nom, int age, int anciennete) {
    this->nom = nom;
    this->age = age;
    this->anciennete = anciennete; 
}

double Employe::calculerSalaire() {
    return calculerBaseSalaire() + (anciennete * 0.10 * calculerBaseSalaire());
}

File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-1_Gestion-Salaires-Employes/Implementation-Java/Employe.java
abstract class Employe {
    protected String nom;
    protected int age;
    protected int anciennete;

    public Employe(String nom, int age, int anciennete) {
        this.nom = nom;
        this.age = age;
        this.anciennete = anciennete;
    }

    public abstract void afficher();

    public abstract double calculerBaseSalaire();

    public double calculerSalaire() {
        return calculerBaseSalaire() + (anciennete * 0.10 * calculerBaseSalaire());
    }
}
